### ISSUE TITLE
GUI applications no longer display after enabling systemd

### ISSUE THREAD

**Issue by kmmiles** [microsoft/WSL#8896]
### Windows build number:

25201.1000

### Your Distribution version:

Debian 11, Ubuntu 20.04.4, Ubuntu 22.04, and Fedora 35

### Your WSL versions:

  WSL version: 0.67.6.0
  Kernel version: 5.15.62.1
  WSLg version: 1.0.44
  MSRDC version: 1.2.3401
  Direct3D version: 1.606.4
  DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
  Windows version: 10.0.25201.1000

### Steps to reproduce:

1. Follow the instructions from the announcement here: https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

2. After modifying `/etc/wsl.conf` to enable `systemd` and restarting with `wsl --shutdown`, try running a GUI program e.g. `xeyes`

3. Disabling `systemd` and restarting eliminates the issue.

### WSL logs:

[pulseaudio.log](https://github.com/microsoft/wslg/files/9629462/pulseaudio.log)
[stderr.log](https://github.com/microsoft/wslg/files/9629467/stderr.log)
[weston.log](https://github.com/microsoft/wslg/files/9629468/weston.log)

I also compared the logs with with `systemd` enabled and disabled, and found something unique in `stderr.log`:

```
  <3>WSLGd: Exception: No such file or directory @../ProcessMonitor.cpp:72 (LaunchProcess)
  <4>WSLGd: /mnt/c/Users/kenne/AppData/Local/Microsoft/WindowsApps/MicrosoftCorporationII.WindowsSubsystemForLinux_8wekyb3d8bbwe/msrdc.exe exited with status
  1.
  <3>WSLGd: Exception: No such file or directory @../ProcessMonitor.cpp:72 (LaunchProcess)
  <4>WSLGd: /mnt/c/Users/kenne/AppData/Local/Microsoft/WindowsApps/MicrosoftCorporationII.WindowsSubsystemForLinux_8wekyb3d8bbwe/msrdc.exe exited with status
  1.
  <3>WSLGd: Exception: No such file or directory @../ProcessMonitor.cpp:72 (LaunchProcess)
  <4>WSLGd: /mnt/c/Users/kenne/AppData/Local/Microsoft/WindowsApps/MicrosoftCorporationII.WindowsSubsystemForLinux_8wekyb3d8bbwe/msrdc.exe exited with status
  1.
  <3>WSLGd: Exception: No such file or directory @../ProcessMonitor.cpp:72 (LaunchProcess)
  <4>WSLGd: /mnt/c/Users/kenne/AppData/Local/Microsoft/WindowsApps/MicrosoftCorporationII.WindowsSubsystemForLinux_8wekyb3d8bbwe/msrdc.exe exited with status   1.
  <3>WSLGd: Exception: No such file or directory @../ProcessMonitor.cpp:72 (LaunchProcess)
  ```
  

### Expected behavior:

![wslg-systemd-expected](https://user-images.githubusercontent.com/7087596/191852262-b5224800-5731-4975-bdaf-a8cb105f552a.png)

### Actual behavior:

![wslg-systemd-actual](https://user-images.githubusercontent.com/7087596/191852330-c7f54ce9-2af9-497f-9ec8-d3fe82ee0da9.png)

**Comment by Anutrix** [comment]
Same issue. I use kali-linux WSL on latest slow-ring/stable available Windows version 22621.963.
WSL version:
```
> wsl --version
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.963
```
Issue seen after systemd was turned on. The following is with systemd enabled but GUI not working:
```
> ls -la /tmp/.X11-unix
total 4
drwxrwxrwx  2 root    root      60 Dec 20 17:02 .
drwxrwxrwt 29 root    root    4096 Dec 20 18:36 ..
srwxrwxrwx  1 xxxxxxx xxxxxxx    0 Dec 20 17:02 X0
```
```
> ls -al /mnt/wslg/.X11-unix/
total 0
drwxrwxrwx 2 root    root     60 Dec 20 17:02 .
drwxrwxrwt 7 root    root    260 Dec 20 17:03 ..
srwxrwxrwx 1 xxxxxxx xxxxxxx   0 Dec 20 17:02 X0
```

**Comment by nullbyto** [comment]
Same problem when trying to start GUI apps on my Ubuntu 20.04.
WSL:
```
> wsl --version
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.2486
```
